community/consistency-check/src/main/java/org/neo4j/consistency/checking/full/CountsBuilderDecorator.java

```diff
@@ -293,9 +293,8 @@ public void check( RelationshipRecord record,
                         }
                     }
                 }
-
-                inner.check( record, engine, records );
             }
+            inner.check( record, engine, records );
```

Comment:
My understanding is that we do multiple passes on the relationship store for checking different constraints. The multipass is needed since we cannot load everything in memory at once. The problem here is that the `CountStoreBuilderDecorator` doesn't want to see the same relationship multiple times, otherwise the counting would be wrong, on the other hand there is no way to decorate only the "first pass" of the consistency checker.

If you look at the check for for `NodeRecord` that's the way it is suppose to be here as well.